ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -403,31 +407,30 @@ class WebSocketService(
   private def emitOffsetTicksAndFilterOutEmptySteps[Pos](startFrom: Option[domain.StartingOffset])
     : Flow[StepAndErrors[Pos, JsValue], StepAndErrors[Pos, JsValue], NotUsed] = {
 
-    type TickTriggerOrStep = Unit \/ StepAndErrors[Pos, JsValue]
-
-    val tickTrigger: TickTriggerOrStep = -\/(())
-    val zeroState: StepAndErrors[Pos, JsValue] = startFrom.cata(
+    val zeroStep: StepAndErrors[Pos, JsValue] = startFrom.cata(
       x => StepAndErrors(Seq(), LiveBegin(AbsoluteBookmark(x.offset))),
       StepAndErrors(Seq(), LiveBegin(LedgerBegin))
     )
     Flow[StepAndErrors[Pos, JsValue]]
-      .map(a => \/-(a): TickTriggerOrStep)
-      .keepAlive(config.heartBeatPer, () => tickTrigger)
-      .scan((zeroState, tickTrigger)) {
-        case ((state, _), -\/(())) =>
-          // convert tick trigger into a tick message, get the last seen offset from the state
+      .map(a => Step(a))
+      .keepAlive(config.heartBeatPer, () => TickTrigger)
+      .scan((zeroStep, TickTrigger: TickTriggerOrStep[Pos])) {
+        case ((_, TickTrigger), TickTrigger) =>
+          // skip all TickTriggers preceding the initial ACS retrieval
+          (zeroStep, TickTrigger)
+        case ((state, _), TickTrigger) =>
+          // convert TickTrigger into a Step, get the last seen offset from the state
           state.step match {
-            case Acs(_) => (ledgerBeginTick, \/-(ledgerBeginTick))
-            case LiveBegin(LedgerBegin) => (ledgerBeginTick, \/-(ledgerBeginTick))
-            case LiveBegin(AbsoluteBookmark(offset)) => (state, \/-(offsetTick(offset)))
-            case Txn(_, offset) => (state, \/-(offsetTick(offset)))
+            case Acs(_) => (state, TickTrigger) // we don't know the offset yet
+            case LiveBegin(LedgerBegin) => (ledgerBeginTick, Step(ledgerBeginTick))
+            case LiveBegin(AbsoluteBookmark(offset)) => (state, Step(offsetTick(offset)))
```

Comment:
This is probably just me being unfamiliar with the code but why do we need to update the state in the case of `LedgerBegin` but not in the case of `AbsoluteBookmark`? As far as I can tell, the only thing that updating the state does here is reset the errors since the step is unchanged anyway.

Reply:
you want to update it for the TickTrigger, you would loose the captured offset. It is just in case of ledgerBegin you have a zero state, it does not matter if you udpate or not... but I like the above idea of capturing just offset instead of the step. I am going to make this change.